### PR TITLE
test(lib/data): schemas for changelog and source urls

### LINKED ISSUE #29942: Centralize data to lib/data for easier contributing
### Describe the proposed change(s).

We have a lot of "crowdsourced" data in the repo, spread through different locations.

For example:
- Monorepos: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/monorepos.ts
- Replacements: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/replacements.ts
- Manual metadata for packages: https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/metadata-manual.ts

We should centralize this into `lib/data` so that it's easier for one-time or occasional contributors to find the right location to edit.

We should keep this folder as "raw" as possible and any wrapper code exists elsewhere. One topic per file. Files should be in .json format.

There should be a readme in the folder which clearly describes what each file/dataset is for.

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Add json schema and ZodSchema for the changelog-urls.json file
- Add json schema and ZodSchema for the source-urls.json file

## Context
Closes: https://github.com/renovatebot/renovate/issues/29942

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: tools/schemas/changelog-urls-schema.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "https://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "patternProperties": {
+    "^[a-zA-Z]+$": {
```
Comment (viceice) [inline tools/schemas/changelog-urls-schema.json:5]: shouldn't we allow more, depNames can have `-/@` too, eg npm

Comment (RahulGautamSingh) [reply tools/schemas/changelog-urls-schema.json:5]: This regex is for the datasource. There's another below which is used for the depNames. I have updated it to include @.

Comment (viceice) [reply tools/schemas/changelog-urls-schema.json:5]: we need `-` for datasource 😁

File: tools/schemas/source-urls-schema.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "https://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "patternProperties": {
+    "^[a-zA-Z]+$": {
```
Comment (viceice) [inline tools/schemas/source-urls-schema.json:5]:
```suggestion
    "^[a-zA-Z-]+$": {
```

File: tools/schemas/changelog-urls-schema.json (hunk 1 above)
Comment (RahulGautamSingh) [inline tools/schemas/changelog-urls-schema.json:5]:
```suggestion
    "^[a-zA-Z-]+$": {
```

**Review by secustor — APPROVED** [review]

**Pull request merged**